### PR TITLE
fft is corrected

### PR DESCRIPTION
**Pull request by subeeshmp**
Added as a new branch.

### CODE REVIEW

File: fft.py [hunk 1]
```diff
@@ -8,40 +9,30 @@
...
 
-import numpy as np
 def npfft(x,t):
     '''Return the normalised amplitude of fft and frequency using fast FFT in Numpy'''
     import numpy as np
     N=x.size
     fft = np.fft.fft(x)
     T = t[1] - t[0]  # sampling interval 
     # 1/T = frequency
-    f = np.linspace(0, 1 / T, N)
```
Comment (p-amol) [inline fft.py:12]: Use numpy.fft.fftfreq to calculate the frequency.

https://het.as.utexas.edu/HET/Software/Numpy/reference/generated/numpy.fft.fftfreq.html

File: fft.py [hunk 2]
```diff
@@ -8,40 +9,30 @@
...
-    e = np.exp(-2j * np.pi * k * n / N)
-    return np.dot(e, x),N
+#     N = x.size
+#     n = np.arange(N)
+#     k = n.reshape((N, 1))
+#     e = np.exp(-2j * np.pi * k * n / N)
+#     return np.dot(e, x),N
 
-import numpy as np
 def npfft(x,t):
```
Comment (p-amol) [inline fft.py]: Instead of t, the input should be delta t.

File: fft.py [hunk 3]
```diff
@@ -8,40 +9,30 @@
...
     '''Return the normalised amplitude of fft and frequency using fast FFT in Numpy'''
     import numpy as np
     N=x.size
     fft = np.fft.fft(x)
     T = t[1] - t[0]  # sampling interval
```
Comment (p-amol) [inline fft.py]: If delta t is input, we can avoid this calculation.

**Review by p-amol — CHANGES_REQUESTED** [review]

File: fft.py (hunk 3 above)
Comment (subeeshmp) [reply fft.py]: But it better to reduce the input right?

File: fft.py (hunk 2 above)
Comment (subeeshmp) [reply fft.py]: Ok

File: fft.py (hunk 3 above)
Comment (p-amol) [reply fft.py]: I meant replace 't' by 'deltat'.  We don't require 't' because FFT assumes data without gap.

File: fft.py [hunk 4]
```diff
@@ -28,15 +28,15 @@
 #=======
 #>>>>>>> 28dff82ecdfe3d39c6bbdf1744410d2af80267a9
 
-def npfft(x,t):
+def npfft(x,delta_t):
     '''Return the normalised amplitude of fft and frequency using fast FFT in Numpy'''
```
Comment (p-amol) [inline fft.py:32]: Just for information. Always make it a habit to write input/output details in comments. Like you did before.

**Pull request merged**